### PR TITLE
Add php_unit_test_case_static_method_calls

### PR DESCRIPTION
**Pull request by franmomu**
This will change all the PHPUnit dynamic calls to static calls and help with the strict rules phpstan plugin

Or do we want to do this when the time comes for each package?

### CODE REVIEW

**Review by phansys — APPROVED** [review]

**Review by OskarStark — APPROVED** [review]

**Review by jordisala1991 — APPROVED** [review]
Do you have time to help with the merges? this could cause some conflicts.

**Comment by franmomu** [comment]
I'm going out and I won't be around until this night I think 😞 I can do it this night or tomorrow morning

File: tests/Action/DetermineNextReleaseVersionTest.php [hunk 1]
```diff
@@ -66,14 +66,14 @@ public function determine(string $expected, string $current, array $pullRequests
...
-        self::assertSame(
+        static::assertSame(
             $expected,
             $nextVersion
         );
     }
 
     /**
-     * @return \Generator<array{0: string, 1: string, 2: array<PullRequest>}>
+     * @return \Generator<array{0: string, 1: string, 2: array<PullRequest>|}>
```
Comment (VincentLanglet) [inline tests/Action/DetermineNextReleaseVersionTest.php:76]: This is weird

**Review by VincentLanglet — APPROVED** [review]

**Review by jordisala1991 — APPROVED** [review]
Merge when you are ready for the hell merge and fixing (I can try to help if im here).

**Comment by franmomu** [comment]
Yeah, I will wait until next week, I'm having some days off, when we do this we can also add the changes from https://github.com/sonata-project/SonataAdminBundle/pull/7436.

**Comment by VincentLanglet** [comment]
Do you plan to do this soon @franmomu ? Do you want some help ?

**Comment by franmomu** [comment]
> Do you plan to do this soon @franmomu ? Do you want some help ?

I'll try today

**Comment by VincentLanglet** [comment]
> > Do you plan to do this soon @franmomu ? Do you want some help ?
> 
> I'll try today

Do you need me to deploy/dispatch the changes or do you have access to the symfony cloud ?

**Pull request merged**